framework/include/utils/PetscSupport.h

```diff
@@ -38,14 +38,23 @@ namespace PetscSupport
 class PetscOptions
 {
 public:
-  PetscOptions() : flags("", "", true) {}
+  PetscOptions()
+    : flags("", "", true), dont_add_these_options("", "", true), user_set_options("", "", true)
+  {
+  }
 
   /// PETSc key-value pairs
   std::vector<std::pair<std::string, std::string>> pairs;
 
   /// Single value PETSc options (flags)
   MultiMooseEnum flags;
 
+  /// Flags to explicitly not set, even if they are specified programmatically
+  MultiMooseEnum dont_add_these_options;
+
+  /// Options that are set by the user at the input level
+  MultiMooseEnum user_set_options;
```

Comment:
To be honest I hate that these are MultiMooseEnums since the PETSc options database is constantly changing and there is no hope of enumerating them all in MOOSE. Additionally, I think a `std::set` or `std::vector` would just be a lot clearer. But I am ok with you adding these new data members, following the existing pattern. Fixing this can be follow-on work by someone